studio/libs/tmp/utils/common.js

```diff
@@ -195,7 +195,10 @@ async function getRequestInfos(context) {
     if (!parsedLocale || !surface || !fragmentPath || !body) {
         const fetchResult = await context.promises?.fetchFragment;
         if (fetchResult) {
-            ({ parsedLocale, surface, fragmentPath, body } = fetchResult);
+            parsedLocale = parsedLocale || fetchResult.parsedLocale;
+            surface = surface || fetchResult.surface;
+            fragmentPath = fragmentPath || fetchResult.fragmentPath;
+            body = body || fetchResult.body;
         }
```

Comment:
if you do this here, you would need to do it in io/www too, and i would recommend not too (as this means io change too)